docker_cli/image.py

```diff
@@ -0,0 +1,458 @@
+        CalledProcessError
+            If the docker build command fails.
+        ValueError
+            If both `dockerfile` and `dockerfile_string` are defined.
+        """
+        ...
+
+    @classmethod
+    def build(
+        cls,
+        tag,
+        *,
+        dockerfile=None,
+        dockerfile_string=None,
+        context=".",
+        stdout=None,
+        stderr=None,
+        network="host",
+        no_cache=True
+    ):
+        if dockerfile is not None and dockerfile_string is not None:
+            raise ValueError(
+                "Both dockerfile and dockerfile_string passed as arguments."
+            )
+
+        # Build with dockerfile if dockerfile_string is None
+        dockerfile_build = dockerfile_string is None
+
+        context_str = os.fspath(".") if context is None else os.fspath(context)
+        cmd = [
+            "docker",
+            "build",
+            f"--network={network}",
+            context_str,
+            "-t",
+            tag
+        ]
+
+        if no_cache:
+            cmd += ["--no-cache"]
+
+        if dockerfile_build:
+            # If a dockerfile path is given, include it.
+            # Else, docker build will default to "./Dockerfile"
+            if dockerfile is not None:
+                cmd += [f"--file={os.fspath(dockerfile)}"]
+            stdin = None
+        else:
+            cmd += ["-f-"]
+            stdin = dockerfile_string
+
+        run(
+            cmd,
+            text=True,
+            stdout=stdout,  # type: ignore
+            stderr=stderr,  # type: ignore
+            input=stdin
+        )
+
+        return cls(tag)
+
+    def _inspect(self, format: Optional[str] = None) -> str:
+        """
+        Use 'docker inspect' to retrieve a piece of information about the
+        image.
+
+        Parameters
+        ----------
+        format : str, optional
+            The value to be requested by the --format argument, or None.
+            Defaults to None.
+
+        Returns
+        -------
+        str
+            The string returned by the docker inspect command.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails.
+        """
+        cmd = ["docker", "inspect"]
+        if format:
+            cmd += [f"-f={format}"]
+        cmd += [self._id]
+
+        inspect_result = run(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+
+        output_text = inspect_result.stdout
+        return output_text
+
+    def run(
+        self,
+        command: str,
+        *,
+        stdout: Optional[Union[io.TextIOBase, int]] = None,
+        stderr: Optional[Union[io.TextIOBase, int]] = None,
+        interactive: bool = False,
+        network: str = "host"
+    ):
+        """
+        Run the given command on a container.
+
+            .. note::
+        For stderr special values to pass into `stdout` and `stdin`, review
```

Comment:
>For stderr special values to pass into `stdout` and `stdin`

Note quite sure what you mean by this. Did you mean to say this instead:

```suggestion
        For subprocess special values to pass into `stdout` and `stderr`, review
```

?

Reply:
Should we move this note down to the description of the `stdout` and `stderr` parameters? I think it's sort of hard to make sense of what you're saying here without the context that's given in the "Parameters" section below.